server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -92,7 +105,10 @@ private void checkCompatibility(MappedFieldType existingFieldType, MappedFieldTy
 
     /** Returns the field for the given field */
     public MappedFieldType get(String field) {
-        return fullNameToFieldType.get(field);
+        String resolvedField = aliasToConcreteName.get(field);
+        return resolvedField == null
+            ? fullNameToFieldType.get(field)
+            : fullNameToFieldType.get(resolvedField);
```

Comment:
things would be a bit easier to read for me this way:

```java
String resolvedField = aliasToConcreteName.getOrDefault(field, field);
return fullNameToFieldType.get(resolvedField);
```

Reply:
Makes sense, will change.